addons/crm/models/crm_lead.py

```diff
@@ -694,12 +694,12 @@ def _handle_won_lost(self, vals):
         leads_leave_lost = Lead
         won_stage_ids = self.env['crm.stage'].search([('is_won', '=', True)]).ids
         for lead in self:
-            if 'stage_id' in vals:
+            if 'stage_id' in vals and lead.active:
                 if vals['stage_id'] in won_stage_ids:
                     if lead.probability == 0:
                         leads_leave_lost |= lead
                     leads_reach_won |= lead
-                elif lead.stage_id.id in won_stage_ids and lead.active:  # a lead can be lost at won_stage
+                elif lead.stage_id.id in won_stage_ids:  # a lead can be lost at won_stage
```

Comment:
not quite sure about this. 
```
if lead.probability == 0:
    leads_leave_lost |= lead
```
will never be reached again as archived lead have probability = 0.
If we move an archived lead (aka a lost lead) to another stage (not won) it should still be lost, indeed.
But if we move a lost lead to won stage on purpose, this lead should somehow be re-activated.
@LNAodoo ?
if should be reactivated : the fix should be to set active = True to the leads that applies before handling won/lost leads.
if not, then we can delete those two lines.

Reply:
@nboulif : 
After discussing with @LNAodoo with the help of @awa-odoo , we suggest the following : 
Instead of all that has been done in this PR, only add a constraint to avoid moving a lead that is archived.

Indeed, this use case is a weird corner case and should not be done. The archived lead should first be restored before being moved to another stage if needed.
if we cannot move a lost lead in won stage anymore, we won't need to modify the ribbon display condition + same for the rainbow man

Reply:
@dbeguin Thank you all for the review and help! :+1: 
I updated the fix :slightly_smiling_face: